test/CMakeLists.txt

```diff
@@ -30,7 +30,7 @@ add_executable(soltest ${sources} ${headers}
     ${libsolidity_sources} ${libsolidity_headers}
     ${libsolidity_util_sources} ${libsolidity_util_headers}
 )
-target_link_libraries(soltest PRIVATE libsolc yul solidity yulInterpreter evmasm devcore ${Boost_PROGRAM_OPTIONS_LIBRARIES} ${Boost_UNIT_TEST_FRAMEWORK_LIBRARIES})
```

Comment:
This is by the way another reason for going ahead with these changes:
Here we use ``Boost_<C>_LIBRARIES`` variables, which happen to be in fact defined by ``FindBoost``, but as a matter of fact that's undocumented, so we can't rely on these to be present anyways (see https://cmake.org/cmake/help/latest/module/FindBoost.html).

Reply:
Why do you think that's undocumented? It is cleary mentioned in the documentation you linked?
I guess you are referring to the fact that it uses a different method for boost > 1.70.
Unfortunately I couldn't find any documentation for that "boost-cmake project" method they mention.

Reply:
Am I blind :-)? I only see ``Boost_<C>_LIBRARY``, not ``Boost_<C>_LIBRARIES`` in the docs...

Reply:
And boost 1.70 won't define either of them, and boost > 1.70 will probably only define ``Boost_<C>_LIBRARY`` (see https://github.com/boostorg/boost_install/commit/cfa8d55250dfc2635e907e42da423e4eb540dee5). There's no such thing as a "boost-cmake project" it seems (maybe not anymore?) - but in boost 1.70 cmake files start to be part of regular boost (see https://github.com/boostorg/boost_install) - that's what caused the original problem.

Reply:
Ah no, I just didn't read properly. You're right, they are not mentioned. But why not use the `Boost_<C>_LIBRARY` version then?

Reply:
Because they are outdated and no longer defined when using boost 1.70 ;-).

Reply:
but you just said 

>  and boost > 1.70 will probably only define Boost_<C>_LIBRARY

Reply:
``!(1.70 > 1.70)`` ;-).

Boost 1.70 exactly will *not* define them - but they realized that that's not a good idea and now have a commit that adds (only) ``Boost_<C>_LIBRARY`` back, but that'll only be in ``1.71`` (unless they release ``1.70.1`` or something like that first).

Reply:
So it's a mess :-D. But since the "modern" way to handle dependencies is imported targets anyways, the best way to deal with this, is to use them and, if they are not present, define them ourselves as a fallback - that'll always work and that's what this PR does ;-).

Reply:
I really need to read :P 
Man, this is a mess.